src/libraries/Microsoft.PowerFx.LanguageServerProtocol/LanguageServer/LanguageServer.cs

```diff
@@ -128,6 +131,44 @@ public void OnDataReceived(string jsonRpcPayload)
             }
         }
 
+        private void HandleCommandExecutedRequest(string id, string paramsJson)
+        {
+            if (id == null)
+            {
+                _sendToClient(JsonRpcHelper.CreateErrorResult(id, JsonRpcHelper.ErrorCode.InvalidRequest));
+                return;
+            }
+
+            Contracts.AssertValue(id);
+            Contracts.AssertValue(paramsJson);
+
+            if (!TryParseParams(paramsJson, out CommandExecutedParams commandExecutedParams))
+            {
+                _sendToClient(JsonRpcHelper.CreateErrorResult(id, JsonRpcHelper.ErrorCode.ParseError));
+                return;
+            }
+
+            if (string.IsNullOrWhiteSpace(commandExecutedParams.Argument))
+            {
+                //TODO: Log argument is null.
+
+                return;
+            }
+
+            switch (commandExecutedParams.Command)
+            {
+                case CommandName.CodeActionApplied:
+                    var codeActionResult = JsonSerializer.Deserialize<CodeActionResult>(commandExecutedParams.Argument);
+                    Contracts.AssertValue(codeActionResult?.ActionResultContext);
+
+                    // TODO: add code extension.
```

Comment:
>// TODO: add code extension. [](http://example.com/codeflow?start=20&length=28)

And this is where I'll hook things up, and see if we can somehow wire back to the ICodeFixHandler?